include/saber/geometry/detail/impl2.hpp

```diff
@@ -0,0 +1,208 @@
+#ifndef SABER_GEOMETRY_DETAIL_IMPL2_HPP
+#define SABER_GEOMETRY_DETAIL_IMPL2_HPP
+
+// std
+#include <array>
+#include <tuple>
+#include <type_traits>
+
+// saber
+#include "saber/geometry/detail/simd.hpp"
+
+namespace saber::geometry::detail {
+
+template<typename T>
+struct Impl2
+{
+    class Scalar
+    {
+    public:
+        // default ctor
+        constexpr Scalar() = default;
+        ~Scalar() = default;
+
+        // alt ctor
+        constexpr Scalar(T inFirst, T inSecond):
+            mTuple(inFirst, inSecond)
+        {
+            
+        }
+
+        template<std::size_t Index>
+        constexpr T Get() const
+        {
+            static_assert(Index < std::tuple_size_v<decltype(mTuple)>, "Provided index out of bounds.");
+            return std::get<Index>(mTuple);
+        }
+
+        constexpr Scalar& operator+=(const Scalar& inRHS)
+        {
+            std::get<0>(mTuple) += std::get<0>(inRHS.mTuple);
+            std::get<1>(mTuple) += std::get<1>(inRHS.mTuple);
+            return *this;
+        }
+
+        constexpr Scalar& operator-=(const Scalar& inRHS)
+        {
+            std::get<0>(mTuple) -= inRHS.Get<0>();
+            std::get<1>(mTuple) -= inRHS.Get<1>();
+            return *this;
+        }
+
+        constexpr Scalar& operator*=(const Scalar& inRHS)
+        {
+            std::get<0>(mTuple) *= inRHS.Get<0>();
+            std::get<1>(mTuple) *= inRHS.Get<1>();
+            return *this;
+        }
+
+        constexpr Scalar& operator/=(const Scalar& inRHS)
+        {
+            std::get<0>(mTuple) /= inRHS.Get<0>();
+            std::get<1>(mTuple) /= inRHS.Get<1>();
+            return *this;
+        }
+
+    private:
+        std::tuple<T,T> mTuple{}; // 2 is assumed
+    };
+
+    class Simd
+    {
+    public:
+        // default ctor
+        constexpr Simd() = default;
+        ~Simd() = default;
+
+        constexpr Simd(T inFirst, T inSecond):
+            mArray{{inFirst, inSecond}} // TRICKY mnfitz 19oct2024: std::array takes a single parameter of a std::initializer_list
+        {
+
+        }
+
+        template<std::size_t Index>
+        constexpr T Get() const
+        {
+            static_assert(Index < std::tuple_size_v<decltype(mArray)>, "Provided index out of bounds.");
+            return mArray[Index];
+        }
+
+        constexpr Simd& operator+=(const Simd& inRHS)
+        {
+            // NOTE: SIMD intrinsic functions lack a constexpr implementation; contaminating our interface
+            // Use std::is_constant_evaluated() to protect our interface so it can remain constexpr
+            // std::is_constant_evaulated() is only available in c++20 or later
+            // Only in c++ 20 or later, can operator+= deliver a constexpr result
```

Comment:
Comment: "Only in c++ 20 or later, can operator+= deliver a _compile-time_ result"